precise/iaf_psc_exp_ps_time_reversal.h

```diff
@@ -0,0 +1,589 @@
+      double_t bisection_step;
+	};
+    
+    // Access functions for UniversalDataLogger -------------------------------
+    
+    //! Read out the real membrane potential
+    double_t get_V_m_() const { return S_.y2_ + P_.E_L_; }
+    double_t get_I_syn_() const { return S_.y1_ex_ + S_.y1_in_; }
+    double_t get_y1_ex_() const { return S_.y1_ex_; }
+    double_t get_y1_in_() const { return S_.y1_in_; }
+    double_t get_y0_() const {return S_.y0_;}
+
+    
+    // ---------------------------------------------------------------- 
+    
+    /**
+     * @defgroup iaf_psc_exp_ps_time_reversal_data
+     * Instances of private data structures for the different types
+     * of data pertaining to the model.
+     * @note The order of definitions is important for speed.
+     * @{
+     */   
+    Parameters_ P_;
+    State_      S_;
+    Variables_  V_;
+    Buffers_    B_;
+    /** @} */
+    
+    //! Mapping of recordables names to access functions
+    static RecordablesMap<iaf_psc_exp_ps_time_reversal> recordablesMap_;
+  };
+  
+inline
+port iaf_psc_exp_ps_time_reversal::send_test_event( Node& target, rport receptor_type, synindex, bool )
+{
+  SpikeEvent e;
+  
+  e.set_sender(*this);
+  //c.check_event(e);
+  return target.handles_test_event( e, receptor_type );
+}
+
+inline
+port iaf_psc_exp_ps_time_reversal::handles_test_event(SpikeEvent &, port receptor_type)
+{
+  if (receptor_type != 0)
+    throw UnknownReceptorType(receptor_type, get_name());
+  return 0;
+}
+
+inline
+port iaf_psc_exp_ps_time_reversal::handles_test_event(CurrentEvent &, port receptor_type)
+{
+  if (receptor_type != 0)
+    throw UnknownReceptorType(receptor_type, get_name());
+  return 0;
+}
+
+inline
+port iaf_psc_exp_ps_time_reversal::handles_test_event(DataLoggingRequest & dlr, 
+				    port receptor_type)
+{
+  if (receptor_type != 0)
+    throw UnknownReceptorType(receptor_type, get_name());
+  return B_.logger_.connect_logging_device(dlr, recordablesMap_);
+}
+
+inline
+void iaf_psc_exp_ps_time_reversal::get_status(DictionaryDatum & d) const
+{
+  P_.get(d);
+  S_.get(d, P_);
+  (*d)[names::recordables] = recordablesMap_.get_list();
+
+}
+
+inline
+void iaf_psc_exp_ps_time_reversal::set_status(const DictionaryDatum & d)
+{
+  Parameters_ ptmp = P_;  // temporary copy in case of errors
+  double_t delta_EL = ptmp.set(d);            // throws if BadProperty
+  State_ stmp = S_;       // temporary copy in case of errors
+  stmp.set(d, ptmp, delta_EL);      // throws if BadProperty
+
+  // if we get here, temporaries contain consistent set of properties
+  P_ = ptmp;
+  S_ = stmp;
+}
+
+
+// inline
+// void iaf_psc_exp_ps_time_reversal::print_stopwatch()
+// {
+//   regime1.print("branch-1 (no-spike) took: ");
+//   regime2.print("branch-2 (no-spike) took: ");
+//   regime3.print("branch-3 (spike) took: ");
+
+// }
+
+inline
+void nest::iaf_psc_exp_ps_time_reversal::spike_test_(const double_t t1)
```

Comment:
This method can be `const`, I believe. Move it to the `cpp` file, since it is only called from the `cpp` file. It is also so complex that there is no point in inlining it. No `const` qualified need on the pass-by-value argument `t1`.